src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -69,7 +70,7 @@ public class KubernetesLauncher extends JNLPLauncher {
 
     private static final Logger LOGGER = Logger.getLogger(KubernetesLauncher.class.getName());
 
-    private boolean launched;
+    private final AtomicBoolean launched = new AtomicBoolean(false);
```

Comment:
This is not OK—the field is not `transient` so it would be persisted in the XStream form of the node config. Granted these agents are usually ephemeral, but they do survive controller restarts and plugin upgrades.

I would suggest restoring it to `boolean` and just make it `volatile` to shut up SpotBugs.